04-onion-routing.md

```diff
@@ -367,6 +368,149 @@ otherwise meets the amount criterion (eg. some other failure, or
 invoice timeout), however if it were to fulfill only some of them,
 intermediary nodes could simply claim the remaining ones.
 
+### Onion Messages
+
+Onion messages have an onion with an alternate `hop_payload`
+format: a `bigsize` followed by a `onionmsg_payload`.  Note that there
+is no legacy format, thus a `bigsize` of 0 means no payload.
+
+An intermediate node expects an `enctlv` which it can decrypt using
+the `blinding` which it is handed along with the onion message.
+
+1. `tlv_stream`: `onionmsg_payload`
+2. types:
+    1. type: 2 (`reply_path`)
+    2. data:
+        * [`point`:`first_node_id`]
+        * [`point`:`blinding`]
+        * [`...*onionmsg_path`:`path`]
+    1. type: 10 (`enctlv`)
+    2. data:
+        * [`...*byte`:`enctlv`]
+
+1. `tlv_stream`: `encmsg_tlvs`
+2. types:
+    1. type: 1 (`padding`)
+    2. data:
+        * [`...*byte`:`pad`]
+    1. type: 4 (`next_node_id`)
+    2. data:
+        * [`point`:`node_id`]
+    1. type: 12 (`next_blinding`)
+    2. data:
+        * [`point`:`blinding`]
+    1. type: 14 (`self_id`)
```

Comment:
I don't think `self_id` is an informative name. I think it should be `path_id` because it identifies the reply path.

Reply:
Sure.